client/src/components/modal-forms/EditPhaseDialog.js

```diff
@@ -17,26 +17,32 @@ const useStyles = makeStyles(() => ({
   },
 }));
 
-export const NewPhaseDialog = ({ onSubmit, onClose, open }) => {
+export const EditPhaseDialog = ({ onSubmit, onClose, open, content, isNew }) => {
   const { openToast } = useToast();
   const classes = useStyles();
-  const initialValues = {
-    phaseName: '',
-    startDate: '',
-    endDate: '',
-  };
+  const initialValues = content
+    ? {
+        phaseName: content.name ?? '',
+        startDate: content.start_date ?? '',
+        endDate: content.end_date ?? '',
+      }
+    : {
+        phaseName: '',
+        startDate: '',
+        endDate: '',
+      };
 
-  const handleCreatePhase = async (phase) => {
+  const handleEditPhase = async (phase) => {
     const phaseJson = {
       name: phase.phaseName,
       start_date: phase.startDate,
       end_date: phase.endDate,
     };
-    const response = await createPhase(phaseJson);
+    const response = await (isNew ? createPhase(phaseJson) : updatePhase(content.id, phaseJson));
```

Comment:
I kept them as different props just in case. Yes, we could absolutely infer action from whether or not there is content, but I figured it was safer to explicitly state whether we're creating or editing. Makes things a bit more obvious when reading it, too.

Reply:
That's fair